src/main/kotlin/org/partiql/scribe/targets/trino/TrinoRewriter.kt

```diff
@@ -149,7 +149,7 @@ public open class TrinoRewriter(val onProblem: ProblemCallback) : PlanRewriter<R
     }
 
     override fun visitRexOpPathKey(node: Rex.Op.Path.Key, ctx: Rel.Type?): PlanNode {
-        if (node.root.type.asNonNullable() !is StructType) {
+        if (node.root.type.asNonAbsent() !is StructType) {
```

Comment:
Changed to use `asNonAbsent` since there are some cases in which `MISSING` would be one of the types in the `union`. 

E.g. assuming `t` is a struct with some nullable struct `flds`, 

`"t"."flds"."a"` would have type `union(<a's type>, missing)`, which would fail this assertion. In addition to `NullType` we need to also omit `MissingType`.

Reply:
In PLK 0.15.0/v1, we won't need to worry about the absent types since they will be removed.